### ISSUE TITLE
Request support for Linkoze/Tuya Wireless Switch Wifi Version

### ISSUE THREAD

**Issue by AndaPlays** [make-all/tuya-local#774]
# Log Message

```
Logger: custom_components.tuya_local.config_flow
Source: custom_components/tuya_local/config_flow.py:49
Integration: Tuya Local (documentation, issues)
First occurred: 2:07:07 AM (6 occurrences)
Last logged: 2:16:08 AM

Device matches loratap_curtain_switch with quality of 50%. DPS: {"updated_at": 1686182825.6465886, "1": "single_click", "2": "single_click"}
Report this to https://github.com/make-all/tuya-local/issues/
```

# Information about DPS mappings

```
{
  "result": {
    "category": "wxkg",
    "functions": [],
    "status": [
      {
        "code": "switch_mode1",
        "lang_config": {
          "click": "Single Click",
          "double_click": "",
          "press": "Long Press"
        },
        "name": "Switch 1",
        "type": "Enum",
        "values": "{\"range\":[\"click\",\"press\"]}"
      },
      {
        "code": "switch_mode2",
        "lang_config": {
          "click": "Single Click",
          "double_click": "Double Click",
          "press": "Long Press"
        },
        "name": "Switch 2",
        "type": "Enum",
        "values": "{\"range\":[\"click\",\"press\"]}"
      },
      {
        "code": "battery_percentage",
        "lang_config": {
          "unit": "%"
        },
        "name": "Battery",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      }
    ]
  },
  "success": true,
  "t": 1686184273186,
  "tid": "c5c0745f059311eea795ce4cada2c678"
}
```

# Product ID
l8yaz4um5b3pwyvf

other Information:
    "model": "LKWSW201",
    "name": "Wireless Switch",
    "product_name": "Wireless Switch",

# Information about how the device functions

Simple Battery powered Switch with two buttons. 
Bought mine from here:
https://de.aliexpress.com/item/1005004683576863.html?spm=a2g0o.order_list.order_list_main.10.1ca75c5fZT6M5Y&gatewayAdapt=glo2

https://www.adaprox.io/products/linkoze-wireless-switch-2-key?variant=41178716700877

**Comment by make-all** [comment]
It would be useful if you can capture the battery reading dp id, which I think will show up in device diagnostics after some time of running.  For now only click and press sensors for the two buttons are implemented.

**Comment by thewan056** [comment]
This doesn't work the way I expected it to so I'm posting my findings.

_TLDR for fixing if possible:_
Either
- Disable detecting the Long press action so that it does nothing and doesn't affect the state Click and Double click, since you cannot determine whether you Long press or not when all the sensors are OFF
- Remove Press 1 and Press 2 sensors since they do nothing anyway.

or

- Try to figure out how to fix Long press to affect Press 1 and Press 2.

Additionally bonus if we could fix the different button presses so that they don't affect unrelated sensors. Would also be nice if the sensor becomes a toggle instead meaning it turns off right after it turns on. They would greatly simplify automations.

_TLDR stop_

Firstly, this is a scene switch, it is expected when you press once, double press or long press a button it does something. When pressing the button, a red led lights up. Double pressing lights it up twice in quick succession. Long pressing activates by holding down a button until the red light turns off, then you let go.

In Tuya Local HA, Click corresponds to pressing the button once. Double Click is pressing the button twice in quick succession. Press, as explained below, does nothing.


There is a WiFi version and a Zigbee version. I have both. Both are detected and work with Tuya Local but you should press button 1 to wake them up from sleep while detecting them. Tuya Local doesn't seem to recognize the device when pressing button 2 to wake them up.
Note that double click was disabled/removed later after it was initially available for the WiFI version. I am guessing that this is probably due to the high latency of WiFi and the fact it goes offline while sleeping, double click could not be determined to work reliably, as it needs to differentiate between double clicking and clicking once multiple times. The Zigbee version retains the double click.

But in the current Tuya Local this is how it works:

**Applies to all button press types (I guess)**

-  The WiFI version becomes unavailable when it sleeps to save battery.
- In unavailable state for the WiFi version when it sleeps, the button changes state of the sensors from Unavailable to ON when you press it.
- The Zigbee version will always be available in HA as long as it has battery so the above doesn't apply. It is either ON or OFF.
- "I guess" is because as you can see below, the Press 1 and Press 2 sensor never changes state and is always OFF as long as available.
- Any action for 1 doesn't affect 2 and vice versa. 1 only affects 1 and 2 only affects 2 respectively.

**For Pressing once**

- Pressing the button once, Click 1 or Click 2 will change state from OFF to ON.
- Pressing the button again while the device remains available in HA, the Click state does not change to OFF if it is ON state. 
- If the Double Click 1 or 2 was in ON state, it will turn OFF. 
- If Double Click was OFF, it stays OFF.
- Press 1 or Press 2 does not change state, it stays OFF regardless.

**For pressing twice (only applies to the Zigbee version)**

- Pressing the button twice in quick succession, Double click 1 or 2 will change state from OFF to ON
- Pressing the button twice again while the device remains available in HA, the Double click state does not change to OFF if it is ON state. 
- If Click 1 or 2 was in ON state, they will switch OFF. 
- If Click was OFF, it stays OFF.
- Press 1 or Press 2 does not change state, it stays OFF regardless.

**For long pressing**

- Long pressing the button, Click 1 or 2 will change to OFF if they were ON.
- If Click was OFF it stays OFF.
- Double Click 1 or 2 will change to OFF if they were ON.
- If Double click was OFF it stays OFF
- Press 1 or Press 2 does not change state, it stays OFF regardless.

The above complicates automations as you have to correspond to multiple sensors off and on states to know whether did you press once or double press or long press plus whether you did it while it was available or not for the WiFI version. You also may need to switch OFF each sensor after they switch ON to use them more effectively, adding further complications.

Example when you press button 1 once, Click 1 switches ON if OFF or Unavailable so you have to switch it OFF right after it switches ON using you automation since pressing it again while the sensor is ON does nothing. You also need to make sure it doesn't affect your automations that use Double click 1 since it will switch OFF Double click 1 if it was ON, probably by using conditions.

Long press is unusable seeing that if everything is OFF it stays OFF, and if any are ON they will switch OFF which may affect any automations that try to use press once and double press. So this effectively leaves the WiFi version with press once only function that heavily limits is usefulness and doesn't justify its price somewhat.

**Comment by thewan056** [comment]
> It would be useful if you can capture the battery reading dp id, which I think will show up in device diagnostics after some time of running. For now only click and press sensors for the two buttons are implemented.

The dp id is 10

**Comment by make-all** [comment]
In the original info you sent, the Tuya docs give possible values as "click", "double_click" and "press".  But in the log you sent, the current value is "single_click".  So the value in the docs for "click" was obviously wrong, and I changed it for the config.  In the absence of any information about the others, I just copied them from the Tuya docs, but it looks like "press" is also wrong. To fix this, you will need to capture the actual value, which you will see in the device diagnostics and maybe in Developer Tools or the log.

Regarding the behaviour, these are buttons not switches, so they aren't supposed to keep a state.  Rather than treating the binary_sensor as a switch and reading its value, you should only trigger automations on the change in state from off or unavailable to on, which will happen when the button is pressed.  It looks like the buttons are retaining their state though, so pressing the same button as the previous one may not be working.   The config already has the "persist: false" attribute set, so they are expected to go unavailable when there is no activity, but if the device keeps telling us the last button pressed, then there is no way to tell when you press it a second time.  Possibly then, these buttons are designed to work in tandom so one button is "on" and the other "off", and turning on when it is already on is expected to be a no-op.  Or they just don't work properly locally, and you'll need to go via the cloud if you need to be able to detect a second press beyond the double-click timeout.

**Comment by make-all** [comment]
I've changed "press" to "long_press" in the config, as a guess as to what the proper value might be.
